util.py

```diff
@@ -55,28 +55,21 @@ def copyFilesFromDir(sourcedir, dest):
 
 def runCmd2(command, with_stdout=False, with_stderr=False, inputtext=None):
 
-    cmd = subprocess.Popen(command, bufsize=1,
-                           stdin=(inputtext and subprocess.PIPE or None),
-                           stdout=subprocess.PIPE,
-                           stderr=subprocess.PIPE,
-                           shell=isinstance(command, str),
-                           close_fds=True)
-
-#     if inputtext:
-#      (out, err) = cmd.communicate(inputtext)
-#         rv = cmd.returncode
-#     else:
-#         (stdout, stderr) = (cmd.stdout, cmd.stderr)
-#         for line in stdout:
-#             out += line
-#         for line in stderr:
-#             err += line
-#         rv = cmd.wait()
-
-    # the above has a deadlock condition.
-    # The following should suffice in all cases
-    (out, err) = cmd.communicate(inputtext)
-    rv = cmd.returncode
+    try:
+        cmd = subprocess.Popen(command, bufsize=1,
+                               stdin=(inputtext and subprocess.PIPE or None),
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               shell=isinstance(command, str),
+                               close_fds=True)
+
+        # We could poll stdout/stderr for commands outputing large amounts
+        # of data, but the following should suffice in all cases
+        (out, err) = cmd.communicate(inputtext)
+        rv = cmd.returncode
+    except ex:
```

Comment:
This is not valid syntax. It needs to be `except Exception as ex:`

Reply:
Good catch, a bit annoyed that python2 allowed me to write that :(